src/utils.js

```diff
@@ -333,19 +325,208 @@ function mapSlideCoordToPixelCoord(options) {
   const affine = buildInverseTransform({
     orientation: options.orientation,
     offset: options.offset,
-    spacing: options.spacing
-  })
+    spacing: options.spacing,
+  });
+
+  return applyInverseTransform({ coordinate: point, affine: affine });
+}
+
+/**
+ * Get coordinate with offset
+ *
+ * @param {object} feature feature
+ * @param {number} offset offset
+ * @returns {array} coordinates with offset
+ */
+function coordinateWithOffset(feature, offset = 70) {
+  const geometry = feature.getGeometry();
+  const coordinates = geometry.getLastCoordinate();
+  const [x, y] = coordinates;
+  return !feature.get(Enums.InternalProperties.Marker) &&
+    feature.get(Enums.InternalProperties.Markup) === Enums.Markup.TextEvaluation
+    ? coordinates
+    : [x - offset, y - offset];
+}
+
+/**
+ * Get view unit suffix
+ *
+ * @param {object} view Map view
+ * @returns {string} unit suffix
+ */
+function getUnitSuffix(view) {
+  const UnitsEnum = { METERS: "m" };
+  const DEFAULT_DPI = 25.4 / 0.28;
+
+  const center = view.getCenter();
+  const projection = view.getProjection();
+  const resolution = view.getResolution();
+
+  const pointResolutionUnits = UnitsEnum.METERS;
+
+  let pointResolution = getPointResolution(
+    projection,
+    resolution,
+    center,
+    pointResolutionUnits
+  );
+
+  const DEFAULT_MIN_WIDTH = 65;
+  const minWidth = (DEFAULT_MIN_WIDTH * DEFAULT_DPI) / DEFAULT_DPI;
+
+  let nominalCount = minWidth * pointResolution;
+  let suffix = "";
+
+  if (nominalCount < 0.001) {
+    suffix = "μm";
+    pointResolution *= 1000000;
+  } else if (nominalCount < 1) {
+    suffix = "mm";
+    pointResolution *= 1000;
+  } else if (nominalCount < 1000) {
+    suffix = "m";
+  } else {
+    suffix = "km";
+    pointResolution /= 1000;
+  }
 
-  return applyInverseTransform({ coordinate: point, affine: affine })
+  return suffix;
 }
 
+/**
+ * Get a num content item instance given value, code, meaning and units
+ *
+ * @param {number} value the measurement value
+ * @param {string} nameCodedConceptValue the coded concept value
+ * @param {string} nameCodedConceptMeaning the value coded concept meaning
+ * @param {string} unitCodedConceptValue the unit coded concept value
+ * @param {string} unitCodedConceptMeaning the unit coded concept meaning
+ * @returns {object} num content item instance
+ */
+const getMeasurementContentItem = (
+  value,
+  nameCodedConceptValue,
+  nameCodedConceptMeaning,
+  unitCodedConceptValue,
+  unitCodedConceptMeaning
+) => {
+  return new dcmjs.sr.valueTypes.NumContentItem({
+    name: new dcmjs.sr.coding.CodedConcept({
+      value: nameCodedConceptValue,
+      meaning: nameCodedConceptMeaning,
+      schemeDesignator: "DCM",
```

Comment:
We cannot assume that the coding scheme will always be "DCM". In fact, for both the "Area" and "Length" measurements, the correct coding scheme is "SCT".

I would suggest removing this level of abstract (i.e., removing the `getMeasurementContentItem` utility) and just using `dcmjs.sr.valueTypes.NumContentItem` directly.

Reply:
See [CID 7470 Linear Measurements](http://dicom.nema.org/medical/dicom/current/output/chtml/part16/sect_CID_7470.html) and [CID 7471 Area Measurements](http://dicom.nema.org/medical/dicom/current/output/chtml/part16/sect_CID_7471.html)